book/homework_1_solutions.ipynb

```diff
@@ -0,0 +1,3405 @@
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<matplotlib.legend.Legend at 0x7f75ec615730>"
+      ]
+     },
+     "execution_count": 45,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create a plot that uses the colors assigned to the table of the Atomic Mass vs the Melting Point.\n",
+    "plt.figure()\n",
+    "plt.xlabel('Atomic Mass')\n",
+    "plt.ylabel('Melting Point')\n",
+    "for state in states:\n",
```

Comment:
You actually don't need a loop here. You can do one plot command:

```
plt.scatter('AtomicMass', 'MeltingPoint', data=df, color="StateMarker")
```

However, doing this doesn't result in an informative legend. Not sure if there would be a way to fix it.

Reply:
Is it worth changing considering the legend becomes harder to understand? I think the only way to preserve an informative legend is to plot them line by line. There might be a solution that uses a legend handler, but I am unsure and think that would be beyond the lesson scope. I believe that leaves us with either a loop or adding each line individually:
```
solid_df = df[df['StandardState'] == 'Solid']
solid_line = state_ax.scatter('AtomicMass', 'MeltingPoint', data=solid_df, color=solid_df.iloc[0]['StateMarker'])
```
And I think the loop is much cleaner.